crates/cairo-lint-core/src/lints/bool_comparison.rs

```diff
@@ -0,0 +1,35 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::ast::{BinaryOperator, Expr, ExprBinary};
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::TypedSyntaxNode;
+
+pub const BOOL_COMPARISON: &str = "Unnecessary comparison with a boolean value. Use the variable directly.";
+
+pub fn check_bool_comparison(db: &dyn SyntaxGroup, node: ExprBinary, diagnostics: &mut Vec<PluginDiagnostic>) {
+    let lhs = node.lhs(db);
+    let op = node.op(db);
+    let rhs = node.rhs(db);
+
+    let is_comparison_operator = matches!(
+        op,
+        BinaryOperator::EqEq(_)
+            | BinaryOperator::Neq(_)
```

Comment:
same here let's only keep those 2 and remove the others

Reply:
@0xLucqs updated!